source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1024,23 +1022,23 @@ bool COpenGLDriver::genericDriverInit()
 	// Reset The Current Viewport
 	glViewport(0, 0, Params.WindowSize.Width, Params.WindowSize.Height);
 
-	glEnable(GL_FRAMEBUFFER_SRGB);
-    glDisable(GL_DITHER);
-    glDisable(GL_MULTISAMPLE);
-    glEnable(GL_TEXTURE_CUBE_MAP_SEAMLESS);
-    extGlClipControl(GL_UPPER_LEFT, GL_ZERO_TO_ONE);
-	glClearDepth(0.0);
-	glDepthFunc(GL_GEQUAL);
-	glDepthRange(1.0,0.0);
-	glFrontFace(GL_CCW);
-
 	// adjust flat coloring scheme to DirectX version
 	///extGlProvokingVertex(GL_FIRST_VERTEX_CONVENTION_EXT);
 
 	// We need to reset once more at the beginning of the first rendering.
 	// This fixes problems with intermediate changes to the material during texture load.
-    //TODO hm? should i worry about that?
-	ResetRenderStates = true;
+    SAuxContext* found = getThreadContext_helper(false);
+    found->nextState.rasterParams.framebufferSRGB = 1;
+    found->nextState.rasterParams.ditherEnable = 0;
+    found->nextState.rasterParams.multisampleEnable = 0;
+    found->nextState.rasterParams.textureCubemapSeamlessEnable = 1;
```

Comment:
@Crisspl multisample enable is always true whenever the pipeline says its using more than 1 sample

Reply:
and vulkan dictates that the pipeline must declare the same number of samples as the framebuffer attachment it will draw to

So yes, OIT example will be broken (Vulkan does not allow drawing to multisample target with multisample disabled to abuse it for OIT) until we expose the user configurable sample locations extension:
- https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VK_EXT_sample_locations
- https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_sample_locations.txt
- https://www.khronos.org/registry/OpenGL/extensions/NV/NV_sample_locations.txt

Reply:
>multisample enable is always true whenever the pipeline says its using more than 1 sample

resolved